.github/workflows/test-linux.yml

```
@@ -31,3 +31,6 @@ jobs:
         run: make build-ci -j2 V=1 CONFIG_FLAGS="--error-on-warn"
       - name: Test
         run: make run-ci -j2 V=1 TEST_CI_ARGS="-p actions"
+      - name: Test Internet
+        # Only run if files in test/internet have changed.
+        run: if [[ $(git diff --name-only origin HEAD -- test/internet) ]]; then make test-internet -j2 V=1; fi
```

```suggestion
        run: |
          if [[ $(git diff --name-only origin HEAD -- test/internet) ]]; then
            make test-internet -j2 V=1;
          fi
```
The only problem is that it will be green if there are no changes in test/internet, so it might not be practical to know if it really ran or not IMO. A solution would be another step before that outputs the exit code of the command, and put an if on the next step.
```yaml
     - run: |
        (git diff --name-only origin HEAD -- test/internet)
        echo "::set-output name=changed::$?"
        id: check_files
      - name: Test Internet
        # Only run if files in test/internet have changed.
        if: 
        run: make test-internet -j2 V=1
```

---

> The only problem is that it will be green if there are no changes in test/internet, so it might not be practical to know if it really ran or not IMO.

I'm OK with that, especially if it keeps the  workflow simple and light on dependencies.

> Refs (-j2 -> -j): #40080 (comment)

I left a comment there but I'm pretty sure we don't want to do that.

---

The multi-line suggestion is a good one. Right now, this isn't working though because of the way the workflow fetches from `origin`.  So that `if` stuff might be going away. If not, I'll do it your way for sure. Thanks! Pondering the right solution...

---

What made you add `--exit-code`?

---

> What made you add --exit-code?

`git diff` always exits with code 0, even if diff is empty.
With --exit-code it exits with code 1 if changes, otherwise 0.

---

I don't think we need that. The `if [[ $( COMMAND ) ]]` in the `if` statement means "true if there is any output from COMMAND, otherwise false".